### ISSUE TITLE
Uncaught TypeError: Cannot set property '_leaflet_pos' of null

### ISSUE THREAD

**Issue by mfc-julius** [Leaflet/Leaflet#3796]
Leaflet 0.7.4 + standalone Mapbox.js + MarkerClusters:

Uncaught TypeError: Cannot set property '_leaflet_pos' of null

```
L.DomUtil.setPosition @ vendor.js:32172
L.Marker.L.Class.extend._setPos @ vendor.js:34779
(anonymous function) @ vendor.js:34681
L.Util.bind @ vendor.js:31193
```

Keep getting this when moving map, if there are markers on it.

FYI anonymous function is, obviously, "this._setPos(pos);" line in:

```js
update: function () {
  if (this._icon) {
    var pos = this._map.latLngToLayerPoint(this._latlng).round();
    L.Util.requestAnimFrame(function () {
      this._setPos(pos);
    }, this);
  }

  return this;
}
```

**Comment by mourner** [comment]
1. Can you set up a minimal JSFiddle reproducing the issue?
2. Why is the issue submitted here and not, say, Leaflet.markercluster repo?

**Comment by mfc-julius** [comment]
Sorry, could open on Leaflet.markercluster, of course, but it does not happen with 0.7.3. That's why I though this might be Leaflet's issue.

Try to move map, zoom in and out: http://jsfiddle.net/svsjujjv/1/

**Comment by mourner** [comment]
@danzel got any ideas?

**Comment by mourner** [comment]
OK I see, there's a race condition caused by https://github.com/Leaflet/Leaflet/commit/c759481592830f80fad59fe7b63d9e14cb406e2f. @perliedman `this._icon` can be unset, we should safeguard against that if we use reqAnimFrame...

**Comment by mourner** [comment]
Confirmed that removing reqAnimFrame fixes the issue: http://jsfiddle.net/svsjujjv/3/
Even if we add a check for `this._icon`, there's still a bug that the markers animate from the top left corner.
We need to find a fix that would both handle the race condition fixed by reqAnimFrame and not have the animation troubles, then release a 0.7.5 regression fix sadly.

**Comment by perliedman** [comment]
@mourner wow, truely sorry about that.

I'm not sure I see the animation problem you're talking about though, at least not in that fiddle - or am I looking for something subtle? The markers and clusters animate pretty much as I would expect them to on Chrome 44 and Firefox 39 on Ubuntu 14.04.

**Comment by mourner** [comment]
@perliedman are you looking at the last fiddle? It's the fixed one. If you comment out the patched update method it bugs out for me on Chrome under OS X.

**Comment by mourner** [comment]
Same animation bug in FF under OS X.

**Comment by perliedman** [comment]
@mourner yes, without the patch I get the "Uncaught TypeError: Cannot set property '_leaflet_pos' of null" and animations look terrible. With your patch, everything looks like expected. I misunderstood it as if the animation problem was something you saw after adding the fix.

**Comment by mourner** [comment]
The fix is just reverting your earlier race condition fix. We need to find a way to keep it while not having the bug somehow...

**Comment by perliedman** [comment]
Ok, so I have done some panic research and found this:

* No obvious way to fix the regression when using `requestAnimFrame`; I'm not too familiar with the internals of Markercluster, but it seems it relies on the position of the marker being updated synchronously and even some pretty nasty hacks I tried didn't seem to fix that, so it seems like a dead end
* Good news (I think): the problem that `requestAnimFrame` attempts to fix _seems to have been fixed anyway in 0.7.4_: even without `requestAnimFrame` in the `update` method, the test case we used _works_ with 0.7.4. Try this fiddle, which uses 0.7.4 with `requestAnimFrame` reverted: http://jsfiddle.net/nqr6m8wu/5/ (normal, working map) - and compare to this fiddle, which uses 0.7.3, obviously also without `requestAnimFrame`: http://jsfiddle.net/nqr6m8wu/4/ (grey map that can't be panned, zoomed, etc.)

My conclusion is that we can revert my bad fix, and it will make Markercluster work again, and #2021 will still be fixed.

It would of course be highly recommended that someone else verifies this, since it's late and I've tried a bit too many things in the last two hours to be really sure of anything.

**Comment by mourner** [comment]
Thanks so much for looking into this Per!
Looks like we're pretty lucky with the fix. :) I wonder what exact commit other than yours fixed the issue.

**Comment by danzel** [comment]
:+1:

**Comment by danzel** [comment]
Having a ```requestAnimFrame``` in Marker.update feels a bit gross, so reverting that sounds nice to me too.

**Comment by perliedman** [comment]
I ran a bisect and found that this commit is the fix for the marker issue, even without `requestAnimFrame`: https://github.com/Leaflet/Leaflet/commit/91477588faa44e2b266d86a5a44cb98490244d39

Good work, @mourner :)

**Comment by mourner** [comment]
Cool! Released 0.7.5 with the fix.
